File: regress/nodejs/simple.ts
const db: ortdb = ort(dbfile);
const ctx: ortctx = db.connect();

const rc: BigInt = ctx.db_foo_insert('a', 2.0, ortns.enm.a, BigInt(ortns.bits.BITF_a));
if (rc < 0)
	return false;

const badobj: ortns.foo|null = ctx.db_foo_get_id(BigInt(3));
if (badobj !== null)
	return false;

const obj: ortns.foo|null = ctx.db_foo_get_id(rc);
if (obj === null)
	return false;
if (obj.obj.id !== rc)
	return false;
if (obj.obj.a !== 'a')
	return false;
if (obj.obj.c !== ortns.enm.a)
	return false;
if (!(obj.obj.d & BigInt(ortns.bits.BITF_a)))
	return false;

return true;

File: regress/rust/simple.rs
use orb::ort;
use std::env;

fn main() {
    let args: Vec<String> = env::args().collect();
    assert_eq!(args.len(), 2);

    let ctx = ort::connect(&args[1]).unwrap();

    let id1 = ctx.db_foo_insert
        (&"testing".to_string(), 1.2, ort::data::Enm::A, 0x01).unwrap();

    let objs1 = ctx.db_foo_get_id(id1).unwrap().unwrap();

    assert_eq!(&objs1.data.a, "testing");
    assert_eq!(objs1.data.c, ort::data::Enm::A);
    assert_eq!(objs1.data.d, 1);
    assert!(&objs1.data.b + f64::EPSILON >= 1.2 && 
            objs1.data.b - f64::EPSILON <= 1.2);
}
